src/main/java/com/playmonumenta/networkrelay/RemotePlayerManager.java

```diff
@@ -339,33 +367,39 @@ public void destOfflineEvent(DestOfflineEvent event) {
 			return;
 		}
 
-		CustomLogger.getInstance().ifPresent(logger -> logger.fine("Unregistering remote shard"));
-		Map<String, RemotePlayer> remotePlayerCopy = new ConcurrentSkipListMap<>(remotePlayers);
-		for (Map.Entry<String, RemotePlayer> playerEntry : remotePlayerCopy.entrySet()) {
+		MMLog.fine("Unregistering shard " + remoteShardName);
+		Map<String, RemotePlayer> remotePlayersCopy = new ConcurrentSkipListMap<>(remotePlayers);
+		for (Map.Entry<String, RemotePlayer> playerEntry : remotePlayersCopy.entrySet()) {
 			RemotePlayer remotePlayer = playerEntry.getValue();
 			unregisterPlayer(remotePlayer.mUuid);
 		}
 		mRemotePlayerShardMapped.remove(remoteShardName);
 	}
 
+	// Player ran a command
+	@EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+	public void playerCommandPreprocessEvent(PlayerCommandPreprocessEvent event) {
+		String command = event.getMessage();
+
+		if (command.startsWith("/pv ")
+			|| command.equals("/pv")
+			|| command.contains("vanish")) {
+			Bukkit.getScheduler().runTask(NetworkRelay.getInstance(), RemotePlayerManager::refreshLocalPlayers);
+		}
+	}
```

Comment:
I agree the `/team` check isn't required here, since we aren't using the player name chat component. We do want to keep the vanish state in sync, however (which I know wasn't in the first attempt to port this).